### PR TITLE
fix: Build platform specific python packages with ci-build-wheel

### PR DESCRIPTION
**Pull request by pyalex**
**What this PR does / why we need it**:

Currently we cannot upload platform specific wheels generated by `wheel`. It produces not-portable wheels (see [PEP-600](https://peps.python.org/pep-0600/)), which are not accepted anymore by PyPi.

This PR reworks our packaging process.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -145,15 +145,15 @@ install-go-ci-dependencies:
 	go get github.com/go-python/gopy
 	go install golang.org/x/tools/cmd/goimports
 	go install github.com/go-python/gopy
+	python -m pip install pybindgen==0.22.0
 
 install-protoc-dependencies:
-	pip install grpcio-tools==1.34.0
+	pip install grpcio-tools==1.44.0 mypy-protobuf
```
Comment (achals) [inline Makefile:151]: need to pin mypy-protobuf?

**Review by achals — APPROVED** [review]
only one question

File: Makefile (hunk 1 above)
Comment (pyalex) [reply Makefile]: pinned

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**